### ISSUE TITLE
Higher Order Components are missing from new documentation

### ISSUE THREAD

**Issue by marckornberger** [reactjs/react.dev#6360]
I found the "Advanced Guides" section in the legacy react docs extremely useful - especially [Higher Order Components (HOCs)](https://legacy.reactjs.org/docs/higher-order-components.html).

Would it be possibly to add this section once again to the new docs? (or possibly on a differnet subpage)

I think the section was extremely useful for react new comers just starting out.

**Comment by michaeljauk** [comment]
Yes, HoCs are extremely useful. Would be very nice, if someone could update the new docs to include them as well. @marckornberger Are you willing to do this?

**Comment by QubitPi** [comment]
I

> I found the "Advanced Guides" section in the legacy react docs extremely useful - especially [Higher Order Components (HOCs)](https://legacy.reactjs.org/docs/higher-order-components.html).
> 
> Would it be possibly to add this section once again to the new docs? (or possibly on a differnet subpage)
> 
> I think the section was extremely useful for react new comers just starting out.

I think they simply won't do with functional component
